### ISSUE TITLE
Make MLFlow into an optional dependency and split pyfunc operators into extras

### ISSUE THREAD

**Issue by timonviola** [astronomer/airflow-provider-mlflow#27]
**Is your feature request related to a problem? Please describe.**
I would like to use this provider, but the MLFlow dependency makes this package bloated.

**Describe the solution you'd like**
I would like to make MLFlow dependency into ```optional``` and make the pyfunc part into an ```extra```. This would allow users to decide if they only need a light weight provider (Registry and Deployments) or if they want the complete MLFlow client - including the pyfunc API.

- basic installation:
```pip install airflow-provider-mlflow```
- with pyfunc
```pip install airflow-provider-mlflow[all]``` or ```pip install airflow-provider-mlflow[pyfunc]```

**Describe alternatives you've considered**
I have considered implementing MLFlow provider package which is strictly compatible with MLFlow REST API. However, this would be more or less a duplication of "Registry" and "Deployments" and it would increase the spread of provider packages e.g.: there are multiple Kafka providers.

**Comment by timonviola** [comment]
I see that #30 was opened due to the same reason

Even better would be to just use `mlflow-skinny`

@fhoda What do you think?

**Comment by fhoda** [comment]
@timonviola I agree, can you update your PR to include mlflow-skinny as the basic install. Thank you!
